node-ui/src/hooks/useInternetComputer.tsx

```diff
@@ -0,0 +1,352 @@
+import { useCallback, useState } from 'react';
+import { randomBytes } from 'crypto';
+import { getOrCreateKeypair } from '../auth/ed25519';
+
+import { DelegationIdentity } from '@dfinity/identity';
+
+import apiClient from '../api';
+import {
+  InternetComputerSignatureMessageMetadata,
+  NodeChallenge,
+  Payload,
+  SignatureMessage,
+  SignatureMessageMetadata,
+  SignData,
+  WalletSignatureData,
+  LoginRequest,
+  LoginResponse,
+  WalletMetadata,
+  WalletType,
+} from '../api/dataSource/NodeDataSource';
+import { ResponseData } from '../api/response';
+import { useNavigate } from 'react-router-dom';
+import { setStorageNodeAuthorized } from '../auth/storage';
+import { useServerDown } from '../context/ServerDownContext';
+import translations from '../constants/en.global.json';
+
+interface RequestNodeDataProps {
+  setErrorMessage: (msg: string) => void;
+}
+
+interface useInternetComputerReturn {
+  ready: boolean;
+  walletSignatureData: WalletSignatureData | null;
+  signData: SignData | null;
+  signMessageAndLogin: (
+    isLogin: boolean,
+    setErrorMessage: (msg: string) => void,
+  ) => void;
+  logout: (setErrorMessage: (msg: string) => void) => void;
+  requestNodeData: ({ setErrorMessage }: RequestNodeDataProps) => void;
+  changeNetwork: (
+    networkId: NetworkId,
+    setErrorMessage: (msg: string) => void,
+  ) => void;
+}
+
+const t = translations.useInternetComputer;
+const possibleNetworks = {
+  production: {
+    url: 'https://identity.ic0.app',
+    cannisterId: 'rdmx6-jaaaa-aaaaa-aaadq-cai',
+  },
+  staging: {
+    url: 'https://beta.identity.ic0.app/',
+    cannisterId: 'fgte5-ciaaa-aaaad-aaatq-cai',
+  },
+};
+
+export type NetworkId = keyof typeof possibleNetworks;
+
+export function useInternetComputer(): useInternetComputerReturn {
+  const navigate = useNavigate();
+  const [ready, setReady] = useState<boolean>(true);
+  const { showServerDownPopup } = useServerDown();
+  const [walletSignatureData, setWalletSignatureData] =
+    useState<WalletSignatureData | null>(null);
+  const [signData, setSignData] = useState<SignData | null>(null);
+
+  const [currentNetwork, setCurrentNetwork] = useState(
+    possibleNetworks['production'],
```

Comment:
Is this like mainnet and testnet or something different?

Reply:
Yes, in ICP context production and staging are considered mainnet and testnet